File: backend/src/main/java/com/moebius/backend/dto/exchange/upbit/UpbitTradeDto.java
package com.moebius.backend.dto.exchange.upbit;

import com.moebius.backend.domain.commons.Change;
import com.moebius.backend.domain.commons.Exchange;
import com.moebius.backend.domain.commons.TradeType;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDateTime;

@Getter
@Setter
@ToString
public class UpbitTradeDto {
	private String id;
	private Exchange exchange;
	private String symbol;
	private TradeType tradeType;
	private Change change;
	private double price;
	private double volume;
	private double prevClosingPrice;
	private double changePrice;
	private LocalDateTime createdAt;
}

File: tracker/src/main/kotlin/com/moebius/tracker/dto/upbit/UpbitTradeDto.kt
package com.moebius.tracker.dto.upbit

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.moebius.backend.domain.commons.Change
import com.moebius.backend.domain.commons.Exchange
import com.moebius.backend.domain.commons.TradeType

@JsonIgnoreProperties(ignoreUnknown = true)
data class UpbitTradeDto(
        @JsonProperty("ty")
        val type: String,
        @JsonProperty("cd")
        val symbol: String,
        @JsonProperty("ttms")
        val tradeTimestamp: Long,
        @JsonProperty("tp")
        val tradePrice: Double,
        @JsonProperty("tv")
        val tradeVolume: Double,
        @JsonProperty("ab")
        val askBid: TradeType,
        @JsonProperty("pcp")
        val prevClosingPrice: Double,
        @JsonProperty("c")
        val change: Change,
        @JsonProperty("cp")
        val changePrice: Double,
        @JsonProperty("sid")
        val sequentialId: Long,
        val exchange: Exchange = Exchange.UPBIT
)